src/nidmm/metadata/functions_addon.py

```diff
@@ -53,6 +51,15 @@
     'self_test':                       { 'codegen_method': 'private', 'method_name_for_documentation': 'self_test',             },  # 'fancy_self_test' Public wrapper that raises
 }
 
+functions_locking = {
```

Comment:
Food for thought:

This is identical across drivers.
It'd be nice if we could single source it.

Reply:
It would be. There's been a couple of things now that would be nice to since source, but not always add.